lib/usb_connection.js

```diff
@@ -196,33 +197,36 @@ USB.Connection.prototype.end = function() {
     // Tell the USB daemon to end all processes active
     // on account of this connection
     Daemon.deregister(this, (err) => {
-      this._close();
       if (err) {
         reject(err);
       } else {
-        resolve();
+        this._close(resolve);
       }
     });
   });
 };
 
 USB.Connection.prototype._close = function(callback) {
+
+  if (typeof callback !== 'function') {
+    callback = function() {};
+  }
+
   if (this.closed) {
-    return callback && callback();
+    setImmediate(callback);
+    return;
   }
 
   this.closed = true;
+
   this.epIn.stopPoll(() => {
     var attempt = () => {
       try {
         this.device.close();
+        callback();
```

Comment:
@johnnyman727 I missed this the first time and now this has already landed, but this is a big no-no. Since the `callback` function is wrapped in the `try`, *any* error that occurs within the `callback` function will now silently be dropped and cause `attempt` to be called again.

What do you think of something like this?

```javascript
try {
  this.device.close()
} catch (_) {
  return setTimeout(attempt, 100)
}

cb(null)
```